infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -116,24 +122,30 @@ private void openTables() {
         }
     }
 
-    public Set<String> getStreamsToReplicate() {
+    /**
+     * This method is for ACTIVE to fetch streams to replicate from LOG_REPLICATION_STREAMS_INFO_TABLE.
+     * If it doesn't exist, we will create info table by reading from registry table.
+     *
+     * @return Set of TableInfo for streams to replicate from ACTIVE
+     */
+    public Set<TableInfo> fetchStreamsToReplicate() {
```

Comment:
I see an issue here, let me know if I am missing something:
in the fetchStreamsToReplicate flow, we read the registry table and have a record of the streamsToReplicate in memory. We don't persist them in this flow.

Now on standby, when we find some new streams, we call `LogReplicationConfig.addStreams()`. This writes the new streams into the streamsInfoTable. And then calls fetchStreamsToReplicate, which doesn't update streamsInfoTable, so now streamsInfoTable now has stale data.